File: solutions/csharp/resistor-color-duo/1/ResistorColorDuo.cs
using System;

public static class ResistorColorDuo
{
    public static int Value(string[] colors)
    {
        string firstv = colors[0];
        string secondv = colors[1];

        testforcolor(firstv);
        testforcolor(secondv);

        var output = testforcolor(firstv) + testforcolor(secondv);
        return int.Parse(output);
    }

    private static string testforcolor(string v)
    {
        var value = v switch
        {
            "black" => "0",
            "brown" => "1",
            "red" => "2",
            "orange" => "3",
            "yellow" => "4",
            "green" => "5",
            "blue" => "6",
            "violet" => "7",
            "grey" => "8",
            "white" => "9",
            _ => "-1",
        };
        return value;
    }
}

File: solutions/javascript/resistor-color-duo/3/resistor-color-duo.js
//
// This is only a SKELETON file for the 'Resistor Color Duo' exercise. It's been provided as a
// convenience to get you started writing code faster.
//

export const decodedValue = (colors) => {
  const firstTwoColors = colors.slice(0,2);
  const digits = firstTwoColors.map(colorToDigit);
  return Number(digits.join(''));
}

const colorToDigit = (color) =>{
  return COLORS.indexOf(color);
}

export const COLORS = [
  "black",
  "brown",
  "red",
  "orange",
  "yellow",
  "green",
  "blue",
  "violet",
  "grey",
  "white",
];